emoji-items.component.ts

```diff
@@ -0,0 +1,141 @@
+        if (!this.getLocalStorageDataService.islLocalStorage()) {
+            console.log('Get data from Github');
+            this.getEmojiService.getData().subscribe(dataFromHttp => {
+                    this.emojiAll = this.compileEmoji(dataFromHttp);
+                    this.getLocalStorageDataService.updateLocalStorage(this.emojiAll);
+                    this.loadPage();
+                }
+            );
+        } else {
+            console.log('Get data from LocalStorage');
+            this.emojiAll = this.getActiveEmoji();
+            this.loadPage();
+        }
+    }
+
+    // Подтягиваем emoji в соответствии со страницей
+    public loadPage() {
+        switch (this.pageName) {
+            case '': // Если главная
+                this.emojiPage = this.getEmojiPage(this.getActiveEmoji(), this.pageNumber, this.emojiPerPage);
+                break;
+            case 'favorite': // Если любимые
+                this.emojiPage = this.getEmojiPage(this.getFavoriteEmoji(), this.pageNumber, this.emojiPerPage);
+                break;
+            case 'deleted': // Если удаленные
+                this.emojiPage = this.getEmojiPage(this.getDeletedEmoji(), this.pageNumber, this.emojiPerPage);
+                break;
+            default:
+                this.emojiPage = this.getEmojiPage(this.getActiveEmoji(), this.pageNumber, this.emojiPerPage);
+        }
+        this.emoji = this.emojiPage.emoji;
+        this.emojiSize = this.emojiPage.totalCount;
+    }
+
+    // Метод, который собирает emoji, и возвращает готовый массив объектов emoji
+    private compileEmoji(data: Object): Array<Emoji> {
+        let index = 0;
+        for (let key in data) {
+            this.emoji.push({
+                id: index,
+                name: key,
+                link: data[key],
+                favorite: false,
+                deleted: false
+            });
+            index++;
+        }
+        return this.emoji;
+    }
+
+    // метод, который возвращает все emoji, кроме удаленных
+    private getActiveEmoji(): Array<Emoji> {
+        return this.getLocalStorageDataService.getLocalStorage().filter(
+            emoji => emoji.deleted === false);
+    }
+
+    // методы, которые возвращает удаленные / любимые emoji, если LocalStorage не пуст,
+    // иначе возвращает всю выборку emoji с гитхаба
+    private getDeletedEmoji(): Array<Emoji> {
+        return this.getLocalStorageDataService.islLocalStorage() ?
+            this.getLocalStorageDataService.getLocalStorage()
+                .filter(emoji => emoji.deleted === true) :
+            this.getActiveEmoji();
+    }
+
+    private getFavoriteEmoji(): Array<Emoji> {
+        return this.getLocalStorageDataService.islLocalStorage() ?
+            this.getLocalStorageDataService.getLocalStorage()
+                .filter(emoji => emoji.favorite === true) :
+            this.getActiveEmoji();
+    }
+
+
+    // метод для поиска emoji
+    public emojiSearch(event): void {
```

Comment:
Type for event

Reply:
Public method between private. Make better methods strcuture, without messy public->private->public (etc).